src/pages/Environments.js

```diff
@@ -647,8 +824,31 @@ const Environments = () => {
           loadData()
         }
       }),
-      loading && spinnerOverlay
-    ])
+      migrateDisk && h(MigratePersistentDiskModal, {
+        disk: migrateDisk,
+        workspaces: _.flow(
+          _.get(migrateDisk.googleProject),
+          _.values,
+          _.filter(({ accessLevel, workspace: { googleProject } }) => {
+            return Utils.canWrite(accessLevel) && _.isEmpty(disksByProject[googleProject])
+          }),
+          _.map('workspace'),
+          _.sortBy(({ name }) => _.lowerCase(name))
+        )(workspaces),
+        onDismiss: () => setMigrateDisk(undefined),
+        onSuccess: () => {
+          setMigrateDisk(undefined)
+          loadData()
+        },
+        onContactSupport: () => contactUsActive.set(true),
+        onDeleteDisk: () => setDeleteDiskId(migrateDisk.id)
+      })
+    ]),
+    (count => count > 0 && h(MigratePersistentDisksBanner, { count }, []))(
+      _.countBy('requiresMigration', disks).true
+    ),
```

Comment:
You explained your reasons, but I still think things like this are less readable/more confusing than storing `count` before the `return` expression.